systems/plants/RigidBodyManipulator.cpp

```diff
@@ -535,9 +535,33 @@ void RigidBodyManipulator::updateCollisionElements(const shared_ptr<RigidBody>&
     } else {
       collision_model->updateElementWorldTransform(*id_iter, body->T);
     }
+    
+    // update the body's contact points    
+    getTerrainPoints(body, body->contact_pts);
   }
 };
 
+void RigidBodyManipulator::getTerrainPoints(int body_id, Eigen::MatrixXd &terrain_points) {
+  getTerrainPoints(bodies[body_id], terrain_points);
+}
+
+void RigidBodyManipulator::getTerrainPoints(const shared_ptr<RigidBody>& body, Eigen::MatrixXd &terrain_points) {
+  // clear matrix before filling it again
+  int num_points = 0;
```

Comment:
Should be size_t (or really `typename Matrix4d::Index`. The += on line 561 will result in a possible loss of data (unlikely, but still let's get rid of the warning that would appear on MSVC).